src/main/java/com/ita/if103java/ims/dao/UserDaoImpl.java

```
@@ -0,0 +1,217 @@
+        this.bCryptPasswordEncoder = new BCryptPasswordEncoder();
+        this.userRowMapper = userRowMapper;
+    }
+
+    private PreparedStatement createStatement(User user, String encryptedPassword, LocalDateTime localDateTime,
+                                              String emailUUID, Connection connection) throws SQLException {
+        PreparedStatement preparedStatement = connection.prepareStatement(Queries.SQL_CREATE_USER, Statement.RETURN_GENERATED_KEYS);
+
+        preparedStatement.setString(1, user.getFirstName());
+        preparedStatement.setString(2, user.getLastName());
+        preparedStatement.setString(3, user.getEmail());
+        preparedStatement.setString(4, encryptedPassword);
+        preparedStatement.setObject(5, user.getRole());
+        preparedStatement.setObject(6, localDateTime);
+        preparedStatement.setObject(7, localDateTime);
+        preparedStatement.setBoolean(8, user.isActive());
+        preparedStatement.setString(9, emailUUID);
+
+        return preparedStatement;
+    }
+
+
+    @Override
+    public User create(User user) {
+        try {
+            LocalDateTime currentDate = LocalDateTime.now();
+            String emailUUID = UUID.randomUUID().toString();
+            String encryptedPassword = bCryptPasswordEncoder.encode(user.getPassword());
+            GeneratedKeyHolder holder = new GeneratedKeyHolder();
+            jdbcTemplate.update(connection -> createStatement(user, encryptedPassword, currentDate, emailUUID, connection), holder);
+            user.setCreatedDate(currentDate);
+            user.setUpdatedDate(currentDate);
+            user.setEmailUUID(emailUUID);
+            user.setId(Optional.ofNullable(holder.getKey())
+                .map(Number::longValue)
+                .orElseThrow(() -> new CRUDException("Error during an user creation: " +
+                    "Autogenerated key is null")));
+            return user;
+        } catch (Exception e) {
```

It is not good to catch all exception. If you expect there some exception, than catch only it.